playwright-e2e/functional/helpers/ApiBuilder.ts

```diff
@@ -0,0 +1,351 @@
+
+interface ReplacementAction {
+  action: string;
+  key: string;
+  value?: any;
+}
+
+export class CcdApiHelper {
+  async axiosRequest<T = any>(
+    requestParams: AxiosRequestConfig
+  ): Promise<AxiosResponse<T> | undefined> {
+    try {
+      return await axiosClient(requestParams);
+    } catch (error: any) {
+      console.error(
+        "Utils %s request error %s",
+        requestParams.url,
+        error.message
+      );
+    }
+  }
+
+  async getUserToken(username: string, password: string): Promise<string> {
+    console.info("Getting User Token");
+
+    const idamClientSecret = process.env.IDAM_CLIENT_SECRET;
+    const redirectUri = `https://div-pfe-${env}.service.core-compute-${env}.internal/authenticated`;
+    const idamCodePath = `/oauth2/authorize?response_type=code&client_id=divorce&redirect_uri=${redirectUri}`;
+
+    const idamCodeResponse = await this.axiosRequest({
+      method: "post",
+      url: idamBaseUrl + idamCodePath,
+      headers: {
+        Authorization: `Basic ${Buffer.from(`${username}:${password}`).toString("base64")}`,
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    });
+
+    console.info("Successfully retrieved IdAM code");
+
+    const idamAuthPath = `/oauth2/token?grant_type=authorization_code&client_id=divorce&client_secret=${idamClientSecret}&redirect_uri=${redirectUri}&code=${idamCodeResponse?.data.code}`;
+
+    const authTokenResponse = await this.axiosRequest({
+      method: "post",
+      url: idamBaseUrl + idamAuthPath,
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    });
+
+    console.info("Successfully retrieved user token");
+    return authTokenResponse?.data.access_token;
+  }
+
+  async getUserId(authToken: string): Promise<string> {
+    console.info("Getting User Id");
+
+    const idamDetailsPath = "/details";
+
+    const userDetailsResponse = await this.axiosRequest({
+      method: "get",
+      url: idamBaseUrl + idamDetailsPath,
+      headers: { Authorization: `Bearer ${authToken}` },
+    });
+
+    console.info("Successfully retrieved User ID");
+    return userDetailsResponse?.data.id;
+  }
+
+  async getServiceToken(): Promise<string> {
+    console.info("Getting Service Token");
+
+    const serviceSecret = process.env.CCD_SUBMIT_S2S_SECRET || "";
+    const s2sBaseUrl = `http://rpe-service-auth-provider-${env}.service.core-compute-${env}.internal`;
+    const s2sAuthPath = "/lease";
+
+    const oneTimePassword = otp.authenticator.generate(serviceSecret);
+
+    const serviceTokenResponse = await this.axiosRequest({
+      url: s2sBaseUrl + s2sAuthPath,
+      method: "post",
+      data: {
+        microservice: "divorce_ccd_submission",
+        oneTimePassword,
+      },
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+
+    console.info("Successfully retrieved service token");
+    return serviceTokenResponse?.data;
+  }
```

Comment:
Consider using finrem_case_orchestration instead of divorce_ccd_submission.

https://github.com/hmcts/civil-service/blob/cb12d6d55148b29196a94539ad1e897fbfcaa046/charts/civil-service/values.aat.template.yaml#L283